src/nvim/tui/tui.c

```diff
@@ -809,24 +809,27 @@ static void fix_terminfo(TUIData *data)
 #define TMUX_WRAP(seq) (inside_tmux ? "\x1bPtmux;\x1b" seq "\x1b\\" : seq)
   // Support changing cursor shape on some popular terminals.
   const char *term_prog = os_getenv("TERM_PROGRAM");
+  const char *vte_version = os_getenv("VTE_VERSION");
 
   if ((term_prog && !strcmp(term_prog, "iTerm.app"))
-      || os_getenv("ITERM_SESSION_ID") != NULL) {
-    // iterm
-    data->unibi_ext.enter_insert_mode = (int)unibi_add_ext_str(ut, NULL,
-        TMUX_WRAP("\x1b]50;CursorShape=1;BlinkingCursorEnabled=1\x07"));
-    data->unibi_ext.enter_replace_mode = (int)unibi_add_ext_str(ut, NULL,
-        TMUX_WRAP("\x1b]50;CursorShape=2;BlinkingCursorEnabled=1\x07"));
-    data->unibi_ext.exit_insert_mode = (int)unibi_add_ext_str(ut, NULL,
-        TMUX_WRAP("\x1b]50;CursorShape=0;BlinkingCursorEnabled=0\x07"));
-  } else {
-    // xterm-like sequences for blinking bar and solid block
+      || os_getenv("ITERM_SESSION_ID") != NULL
+      || os_getenv("XTERM_VERSION") != NULL
```

Comment:
`!= NULL` could be removed for consistency with the other `os_getenv` checks in this function (or should be added to the other checks).